src/app/GitUI/Models/OutputHistoryModel.cs

```diff
@@ -0,0 +1,122 @@
+using System.Diagnostics;
+using System.Text;
+using GitUI.Editor.Diff;
+using ICSharpCode.TextEditor.Document;
+
+namespace GitUI.Models;
+
+public interface IOutputHistoryModel
+{
+    event EventHandler HistoryChanged;
+
+    bool Enabled { get; }
+    string History { get; }
+
+    void Clear();
+    void Trace(in RunProcessInfo runProcess);
+    void Trace(in Exception exception);
+    void Trace(in string message);
+}
+
+public sealed class OutputHistoryModel : IOutputHistoryModel
```

Comment:
> I'm having troubles reconciling the name with the way it's used, it just doesn't fit my mental model:
> We call it a "model" but we use it as "service" which performs actions. IMO it should be called in a way that denotes actions. E.g., something like IGitOutputLogHistoryProvider.
> ![image](https://private-user-images.githubusercontent.com/4403806/342029274-397d12c0-67c2-41fc-b104-e1e55b0377f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0Mjc0NDgsIm5iZiI6MTcxOTQyNzE0OCwicGF0aCI6Ii80NDAzODA2LzM0MjAyOTI3NC0zOTdkMTJjMC02N2MyLTQxZmMtYjEwNC1lMWU1NWIwMzc3ZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjZUMTgzOTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDRlMzU2YTcxMGRmYWE0YjkxNWU4ZDU4YzQ1YjZiYmQ4ZWMwZjQzYThkMTFkMzExMjJkMTRhNzYwYzg3MGZlNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.8Yft-Nk9hQH-0MM86KyPdTThJ8SyLUn2lk8wMLnomhU)

It is a model. It manages and formats data by means of actions, properties and a change event. It does not retrieve the data, what a provider would do.
`AddService`/`GetRequiredService`: You chose the term "service" for DI. Perhaps it is too specific.

I am renaming the methods as suggested (although I prefer to avoid the repetition of terms - `History` in this case). So the above code will become clearer.